src/main/kotlin/io/zerobase/smarttracing/resources/OrganizationsResource.kt

```diff
@@ -1,47 +1,150 @@
+    @JsonProperty("email") val email: String,
+    @JsonProperty("contactName") val contactName: String
+)
 
-//endregion
+data class Location(
+    @JsonProperty("latitude") val latitude: Float,
+    @JsonProperty("longitude") val longitude: Float
+)
+
+data class CreateOrganizationRequest(
+    @JsonProperty("name") val name: String,
+    @JsonProperty("contact") val contact: Contact,
+    @JsonProperty("address") val address: String,
+    @JsonProperty("hasTestingFacilities") val hasTestingFacilities: Boolean?,
+    @JsonProperty("hasMultipleSites") val hasMultipleSites: Boolean?
+)
+
+data class CreateSiteRequest(
+    @JsonProperty("name") val name: String?,
+    @JsonProperty("category") val category: String,
+    @JsonProperty("subcategory") val subcategory: String,
+    @JsonProperty("location") val location: Location,
+    @JsonProperty("isTesting") val isTesting: Boolean,
+    @JsonProperty("contact") val contact: Contact?
+)
+
+data class CreateScannableRequest(
+    @JsonProperty("type") val type: String,
+    @JsonProperty("singleUse") val singleUse: Boolean
+)
 
 @Path("/organizations")
 @Consumes(MediaType.APPLICATION_JSON)
 @Produces(MediaType.APPLICATION_JSON)
-class OrganizationsResource {
+@SuppressFBWarnings("RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE", justification = "false positive")
+class OrganizationsResource(val dao: GraphDao, private val siteTypes: MultiMap<String, String>, private val scanTypes: List<String>) {
 
     @POST
     @Creator
-    fun createOrganization(request: Any): IdWrapper {
-        TODO("not implemented")
+    fun createOrganization(request: CreateOrganizationRequest): IdWrapper? {
+        val name = request.name
+        val phone = request.contact.phone
+        val email = request.contact.email
+        val contactName = request.contact.contactName
+        val address = request.address
+        val hasTestingFacilities = request.hasTestingFacilities ?: false
+        val hasMultipleSites = request.hasMultipleSites ?: true
+
+        val id = dao.createOrganization(
+            name, phone, email,
+            contactName, address,
+            hasTestingFacilities,
+            hasMultipleSites
+        )
+
+        return when (id) {
+            null -> null
+            else -> IdWrapper(id)
+        }
     }
 
     @Path("/{id}/sites")
     @POST
     @Creator
-    fun createSite(@PathParam("id") id: String, request: Any): IdWrapper {
-        TODO("not implemented")
+    fun createSite(@PathParam("id") id: String, request: CreateSiteRequest): IdWrapper? {
+        val name = request.name ?: ""
+        val category = request.category
+        val subcategory = request.subcategory
+        val latitude = request.location.latitude
+        val longitude = request.location.longitude
+        val isTesting = request.isTesting
+        val isCat = siteTypes.containsKey(category)
+        val isSubCat = siteTypes[category]?.contains(subcategory)
+
+        if (!(isCat!!)) {
```

Comment:
if you want to check if it's null, do a null check. this is not idiomatic by any stretch of the imagination. also, returning null isn't really a great option here because all that will do is send a 204 to the client with no information about what when wrong or even that something did go wrong.

Reply:
there was a weird error i was getting when i was checking if it was a null, i do not know if kotlin is doing short circuiting of boolean expressions correctly